### PR TITLE
add argo workflows runner

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

- Add new runner for Argo Workflows
- Add check `DefaultServiceAccount` to ensure the `default` ServiceAccount is not used
- Add check `CKV_ARGO_1` and `CKV_ARGO_2`

### Description
By default all pods/containers in a workflow run with the default service account. This gives each pod the ability to communicate with Kubernetes API and easily become over-permissioned.

### Fix
Create a custom service account with least privileged and add it to the spec.
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: example-
spec:
  entrypoint: main
+ serviceAccountName: <specific service account name>
  ...
```

### Description
Same as here https://docs.bridgecrew.io/docs/bc_k8s_22 🙂 

### Fix
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: example-
spec:
  entrypoint: main
  securityContext:
+   runAsNonRoot: true
+   runAsUser: <specific user>
  ...
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/argo_workflows/runner.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from checkov.yaml_doc.runner import Runner as YamlRunner
+
+# Import of the checks registry for a specific resource type
+from checkov.argo_workflows.checks.registry import registry as template_registry
+
+if TYPE_CHECKING:
+    from checkov.common.checks.base_check_registry import BaseCheckRegistry
+
+
+class Runner(YamlRunner):
```
Comment (schosterbarak) [inline checkov/argo_workflows/runner.py:16]: I think it should inherit from Image reference too

**Comment by schosterbarak** [comment]
please update:
https://github.com/bridgecrewio/checkov#readme
https://github.com/bridgecrewio/checkov/tree/master/docs/6.Contribution
https://github.com/bridgecrewio/checkov/tree/master/docs/7.Scan%20Examples

File: checkov/argo_workflows/runner.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return self.block_type_registries["template"]
+
+    def _parse_file(
+        self, f: str
+    ) -> tuple[dict[str, Any] | list[dict[str, Any]], list[tuple[int, str]]] | None:
+        if not f.endswith((".yaml", ",yml")):
+            return None
+
+        content = Path(f).read_text()
+        if "apiVersion" in content and "argoproj.io/" in content:
```
[the remark this answers is not in the document]
Comment (nimrodkor) [reply checkov/argo_workflows/runner.py:36]: @gruebel Can the `argoproj.io` be anywhere in the `content` for real?

File: checkov/argo_workflows/checks/base_argo_workflows_check.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            name=name,
+            id=id,
+            categories=categories,
+            supported_entities=supported_entities,
+            block_type=block_type,
+        )
+        self.path = path
+        registry.register(self)
+
+    def scan_entity_conf(self, conf: dict[str, Any], entity_type: str) -> tuple[CheckResult, dict[str, Any]]:  # type:ignore[override]  # return type is different than the base class
```
Comment (nimrodkor) [inline checkov/argo_workflows/checks/base_argo_workflows_check.py:34]: @gruebel Why? Can we avoid it?
> return type is different than the base class

**Review by nimrodkor — COMMENTED** [review]
Cool @gruebel !

File: checkov/argo_workflows/checks/base_argo_workflows_check.py (hunk 3 above)
Comment (gruebel) [reply checkov/argo_workflows/checks/base_argo_workflows_check.py]: I tried my best, but now it is a result of our `multi_signature` decorator. I still can't make the typing work correctly 😞

File: checkov/argo_workflows/runner.py (hunk 1 above)
Comment (gruebel) [reply checkov/argo_workflows/runner.py]: done

**Comment by gruebel** [comment]
@schosterbarak @nimrodkor code and documentation wise everything should be as asked for 🙂

**Review by tronxd — APPROVED** [review]
🥇

**Review by schosterbarak — APPROVED** [review]

**Pull request merged**